packages/eslint-plugin/src/rules/prefer-optional-chain-utils/gatherLogicalOperands.ts

```diff
@@ -94,7 +95,7 @@ function isValidFalseBooleanCheckType(
     }
   }
 
-  if (options.requireNullish === true) {
+  if (options.requireNullish === true && operatorNode.left === node) {
```

Comment:
This won't work for logical expressions with >2 operators, like for example `foo && foo.bar && foo.bar.toString()`. I haven't dived into the code yet, but I don't think we can rely on `operatorNode.left === node`.

[playground for this pr](https://deploy-preview-8559--typescript-eslint.netlify.app/play/#ts=5.3.3&fileType=.ts&code=CYUwxgNghgTiAEYD2A7AzgF3gMyUgXPAN4BGshmMAligOYC%2BAUI7kvAGTs54B0ZMHLqz6weGJAGUM1OgAoAlEA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Y6RAM0WloHsalfkwCG8WmQAWo5ujABtcNhy9o-aJAA0ylVhXYYiAI6xKnAHIJGyKfPxxEu7AF9nAXWVuXQA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYwkJemy0eAcgK6qoDCAFutANZlKdFugC%2BIUUA&tokens=false)

```ts
declare const foo: {bar: string}
foo && foo.bar && foo.bar.toString()
```

This code should not be reported when `requireNullish` option is enabled, because there are no nullish types here at all. But it's actually reported.

Reply:
good catch, fixed!